packages/inter-protocol/src/vaultFactory/vaultDirector.js

```diff
@@ -196,450 +115,435 @@ const initState = (
-
-        const managerStorageNode =
-          storageNode &&
-          E(storageNode).makeChildNode(`manager${state.managerCounter}`);
-        state.managerCounter += 1;
-
-        /** a powerful object; can modify parameters */
-        const vaultParamManager = makeVaultParamManager(
-          makeStoredPublisherKit(managerStorageNode, marshaller, 'governance'),
-          initialParamValues,
-        );
-        vaultParamManagers.init(collateralBrand, vaultParamManager);
+    });
+  };
 
-        const { timerService } = zcf.getTerms();
-        const startTimeStamp = await E(timerService).getCurrentTimestamp();
+  const updateShortfallReporter = async () => {
+    const oldInvitation = baggage.has(shortfallInvitationKey)
+      ? baggage.get(shortfallInvitationKey)
+      : undefined;
+    const newInvitation = directorParamManager.getInternalParamValue(
+      SHORTFALL_INVITATION_KEY,
+    );
 
-        /**
-         * We provide an easy way for the vaultManager to add rewards to
-         * the rewardPoolSeat, without directly exposing the rewardPoolSeat to them.
-         *
-         * @type {MintAndReallocate}
-         */
-        const mintAndReallocate = (toMint, fee, seat, ...otherSeats) => {
-          const kept = AmountMath.subtract(toMint, fee);
-          debtMint.mintGains(harden({ Minted: toMint }), mintSeat);
-          try {
-            rewardPoolSeat.incrementBy(
-              mintSeat.decrementBy(harden({ Minted: fee })),
-            );
-            seat.incrementBy(mintSeat.decrementBy(harden({ Minted: kept })));
-            zcf.reallocate(rewardPoolSeat, mintSeat, seat, ...otherSeats);
-          } catch (e) {
-            console.error('mintAndReallocate caught', e);
-            mintSeat.clear();
-            rewardPoolSeat.clear();
-            // Make best efforts to burn the newly minted tokens, for hygiene.
-            // That only relies on the internal mint, so it cannot fail without
-            // there being much larger problems. There's no risk of tokens being
-            // stolen here because the staging for them was already cleared.
-            debtMint.burnLosses(harden({ Minted: toMint }), mintSeat);
-            throw e;
-          } finally {
-            // Note that if this assertion may fail because of an error in the
-            // try{} block, but that error won't be thrown because this executes
-            // before the catch that rethrows it.
-            assert(
-              Object.values(mintSeat.getCurrentAllocation()).every(a =>
-                AmountMath.isEmpty(a),
-              ),
-              X`Stage should be empty of Minted`,
-            );
-          }
-          facets.machine.updateMetrics();
-        };
+    if (newInvitation === oldInvitation) {
+      assert(
+        shortfallReporter,
+        'updateShortFallReported called with repeat invitation and no prior shortfallReporter',
```

Comment:
```suggestion
      !shortfallReporter ||
        Fail`updateShortFallReported called with repeat invitation and no prior shortfallReporter`;
```

I guess this is encouraged when the error message has to be computed, which isn't the case here. I'm getting used to this style, but YMMV.